### PR TITLE
WRO-2578: Updated moonstone samples dependencies and migrated to new React style APIs

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Updated moonstone samples dependencies and migrated to new React style APIs

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-2578

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

File: moonstone/pattern-activity-panels-deep-linking/src/store/store.js [hunk 1]
```diff
@@ -1,19 +1,30 @@
-import {createStore} from 'redux';
+import {configureStore, createSlice} from '@reduxjs/toolkit';
 
-import rootReducer from '../reducers';
+const pathSlice = createSlice({
+	name: 'pathReducer',
+	initialState: {
+		path: '/first/second'
+	},
```
Comment (adrian-cocoara-lgp) [inline moonstone/pattern-activity-panels-deep-linking/src/store/store.js:8]: delete empty space

File: moonstone/pattern-activity-panels-redux/src/store/store.js [hunk 2]
```diff
@@ -1,19 +1,33 @@
-import {createStore} from 'redux';
+import {configureStore, createSlice} from '@reduxjs/toolkit';
 
-import rootReducer from '../reducers';
+const indexSlice = createSlice({
+	name: 'indexReducer',
+	initialState: {
+		index: 0
+	},
```
Comment (adrian-cocoara-lgp) [inline moonstone/pattern-activity-panels-redux/src/store/store.js:8]: deelte empty line

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

**Review by vladut-clapou-lgp — APPROVED** [review]
LGTM

**Pull request merged**